tests/manage/z_cluster/nodes/test_non_ocs_taint_and_toleration.py

```diff
@@ -111,50 +116,50 @@ def test_non_ocs_taint_and_tolerations(self):
+            namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+            resource_name=constants.ROOK_OPERATOR_CONFIGMAP,
+        )
+
+        params = (
+            '{"data": {"CSI_PLUGIN_TOLERATIONS": "\n- effect: NoSchedule\n  key: xyz\n  operator: Equal\n  '
+            'value: \\"true\\"", "CSI_PROVISIONER_TOLERATIONS": "\n- effect: NoSchedule\n  key: xyz\n  operator: '
+            'Equal\n  value: \\"true\\""}} '
+        )
+        params = params.replace("\n", "\\n")
+
+        configmap_obj.patch(params=params, format_type="merge")
+        logger.info(f"Successfully added toleration to {configmap_obj}")
 
         # Add tolerations to the ocsinitializations.ocs.openshift.io
         param = (
             '{"spec":  {"tolerations": '
             '[{"effect": "NoSchedule", "key": "xyz", "operator": "Equal", '
             '"value": "true"}]}}'
         )
-
         ocsini_obj = ocp.OCP(
             resource_name=constants.OCSINIT,
             namespace=defaults.ROOK_CLUSTER_NAMESPACE,
             kind=constants.OCSINITIALIZATION,
         )
         ocsini_obj.patch(params=param, format_type="merge")
-
-        # Add tolerations to the configmap rook-ceph-operator-config
-        configmap_obj = ocp.OCP(
-            kind=constants.CONFIGMAP,
-            namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
-            resource_name=constants.ROOK_OPERATOR_CONFIGMAP,
-        )
-        toleration = configmap_obj.get().get("data").get("CSI_PLUGIN_TOLERATIONS")
-        toleration += (
-            '\n- key: xyz\n  operator: Equal\n  value: "true"\n  effect: NoSchedule'
-        )
-        toleration = toleration.replace('"', '\\"').replace("\n", "\\n")
-        param_cmd = (
-            f'[{{"op": "replace", "path": "/data/CSI_PLUGIN_TOLERATIONS", "value": "{toleration}" }}, '
-            f'{{"op": "replace", "path": "/data/CSI_PROVISIONER_TOLERATIONS", "value": "{toleration}" }}]'
-        )
-        configmap_obj.patch(params=param_cmd, format_type="json")
+        logger.info(f"Successfully added toleration to {ocsini_obj}")
 
         # After edit noticed few pod respins as expected
         assert wait_for_pods_to_be_running(timeout=600, sleep=15)
 
-        # Respin all pods and check it if is still running
-        pod_list = get_all_pods(
-            namespace=defaults.ROOK_CLUSTER_NAMESPACE,
-        )
-        for pod in pod_list:
-            pod.delete(wait=False)
+        # Check non ocs toleration on pods under openshift-storage
+        check_toleration_on_pods(toleration_key="xyz")
 
-        assert wait_for_pods_to_be_running(timeout=600, sleep=15)
-        self.sanity_helpers.health_check()
+        # check number of pods before and after adding non ocs taint
+        number_of_pods_after = len(
+            get_all_pods(namespace=defaults.ROOK_CLUSTER_NAMESPACE)
+        )
+        assert number_of_pods_before == number_of_pods_after
 
         # Add capacity to check if new osds has toleration
```

Comment:
Also check that the osd-prepare pods has the expected tolerations in their yaml

Reply:
check_toleration_on_pods(toleration_key="xyz")
This function is checking the toleration for all pods in openshift-storage

Reply:
There is no guarantee that the osd-prepare pods will exist in line 158 as they will disappear if the nodes were rebooted. It is better to check the toleration on the osd-prepare pods after doing the add-capacity seperately.

Reply:
Instead of checking toleration separatly for osd-prepare pod I checked if osd-prepare-pod is present or not after add-capacity . And then checked toleration on all pods